agent/config/config.go

```diff
@@ -390,6 +390,26 @@ type ServiceAddress struct {
 	Port    *int    `mapstructure:"port"`
 }
 
+type ServicePorts []ServicePort
+
+type ServicePort struct {
+	Name    *string `mapstructure:"name"`
+	Port    *int    `mapstructure:"port"`
+	Default *bool   `mapstructure:"default"`
+}
+
+func (sp ServicePort) Validate() error {
```

Comment:
To avoid making copies of the ServicePort

```suggestion
func (sp *ServicePort) Validate() error {
```